generate.go

```diff
@@ -192,3 +221,71 @@ func generateIndex(configType string) (map[string]typeInfo, error) {
 	}
 	return tInfo, nil
 }
+
+// Given a path to config
+func downloadDataFiles(dataFileCfgs ...dataFileConfig) {
+	for _, dataFileCfg := range dataFileCfgs {
+		fmt.Println("Downloading file from " + dataFileCfg.SourceURL)
+		resp, err := http.DefaultClient.Get(
+			dataFileCfg.SourceURL,
+		)
+		if err != nil {
+			log.Fatalf("Unable to download %s : %v", dataFileCfg.SourceURL, err)
+		}
+
+		p := path.Join("./", dataFileCfg.DestFile)
+		dir, _ := path.Split(p)
+
+		err = os.MkdirAll(dir, 0700)
+		if err != nil {
+			log.Fatalf("Unable to create directory: %v", err)
+		}
+		data, err := io.ReadAll(resp.Body)
+		if err != nil {
+			log.Fatalf("Unable to read response body: %v", err)
+		}
+		dfName := getDataFileNameFromPath(p)
+		if _, ok := postDfDownloadProcessors[dfName]; ok {
+			data, err = postDfDownloadProcessors[dfName](data)
+			if err != nil {
+				log.Fatalf("Unable to process %s: %v", p, err)
+			}
+		}
+		err = os.WriteFile(p, data, 0644)
+		if err != nil {
+			log.Fatalf("Unable to write to file: %v", err)
+		}
+	}
+}
+
+func getDataFileNameFromPath(p string) string {
+	splits := strings.Split(p, "/")
+	n := len(splits)
+	author := splits[n-2]
+	name := strings.Split(splits[n-1], ".")[0]
+	return author + "/" + name
+}
+
+func fastlyDfProcessor(data []byte) ([]byte, error) {
```

Comment:
i think we want to have the generation/formating of list in a different binary/code that can be run without the index.json generation